test/integration/controller/job/job_controller_test.go

```diff
@@ -1027,4 +1028,76 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
 		})
 	})
 
+	ginkgo.It("Should schedule jobs when partial admission is enabled", func() {
+		origPartialAdmission := features.Enabled(features.PartialAdmission)
+		ginkgo.By("enable partial admission", func() {
+			gomega.Expect(features.SetEnable(features.PartialAdmission, true)).To(gomega.Succeed())
+		})
+
+		prodLocalQ = testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+		job1 := testingjob.MakeJob("job1", ns.Name).
+			Queue(prodLocalQ.Name).
+			Parallelism(5).
+			Completions(6).
+			Request(corev1.ResourceCPU, "2").
+			Obj()
+		jobKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
+		wlKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(job1.Name), Namespace: job1.Namespace}
+
+		ginkgo.By("creating localQueues")
+		gomega.Expect(k8sClient.Create(ctx, prodLocalQ)).Should(gomega.Succeed())
+
+		ginkgo.By("creating the job")
+		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
+
+		createdJob := &batchv1.Job{}
+		ginkgo.By("the job should stay suspended", func() {
+			gomega.Consistently(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, jobKey, createdJob)).Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.ConsistentDuration, util.Interval).Should(gomega.Equal(pointer.Bool(true)))
+		})
+
+		ginkgo.By("enable partial admission", func() {
+			gomega.Expect(k8sClient.Get(ctx, jobKey, createdJob)).Should(gomega.Succeed())
+			if createdJob.Annotations == nil {
+				createdJob.Annotations = map[string]string{
+					workloadjob.JobMinParallelismAnnotation: "1",
+				}
+			} else {
+				createdJob.Annotations[workloadjob.JobMinParallelismAnnotation] = "1"
+			}
+
+			gomega.Expect(k8sClient.Update(ctx, createdJob)).Should(gomega.Succeed())
+		})
+
+		wl := &kueue.Workload{}
+		ginkgo.By("the job should be unsuspended with a lower parallelism", func() {
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, jobKey, createdJob)).Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(pointer.Bool(false)))
+			gomega.Expect(*createdJob.Spec.Parallelism).To(gomega.BeEquivalentTo(2))
+
+			gomega.Expect(k8sClient.Get(ctx, wlKey, wl)).To(gomega.Succeed())
+			gomega.Expect(wl.Spec.PodSets[0].MinCount).ToNot(gomega.BeNil())
+			gomega.Expect(*wl.Spec.PodSets[0].MinCount).To(gomega.BeEquivalentTo(1))
+		})
+
+		ginkgo.By("changing the min parallelism the job should be suspended and its parallelism restored", func() {
```

Comment:
the controller integration tests run without web-hooks, in a full setup this is restricted.

Reply:
the purpose is to check that the the restore is done properly, not to go around the web-hook.

I chose this because besides suspending the it will also block it's potential re-admission.